json/DataTypes/DecodeStudent.roc

```diff
@@ -0,0 +1,28 @@
+module [readStudents]
+
+import Student exposing [Student, Module]
+import Decoding exposing [JsonDecoder, list, field, string, number, bool, map2, map3, or]
+
+Students : List Student
+
+readStudents : JsonDecoder Students
+readStudents = \json ->
+    studentsField = \s -> (list readStudent) s
+    (field "students" studentsField) json
+
+readStudent : JsonDecoder Student
+readStudent = \json ->
+    nameField = field "name" string
+    modulesField = field "modules" (\m -> (list readModule) m)
+    currentGrade = field "currentGrade" number
+    finalGrade = field "finalGrade" number
+    currentStudent = map3 nameField modulesField currentGrade (\(name, mods, cg) -> CurrentStudent { name: name, modules: mods, currentGrade: cg })
```

Comment:
:art: I'd personally split this one across two lines:

```suggestion
    currentStudent = map3 nameField modulesField currentGrade \(name, mods, cg) ->
        CurrentStudent { name: name, modules: mods, currentGrade: cg }
```

Same with `graduatedStudent` on the next line!